src/actions/taxonomy.js

```diff
@@ -29,12 +48,70 @@ function* loadTaxonomyVocabulary() {
       },
     });
   } catch (error) {
-    yield put(setErrorMessage(error.toString()));
+    const errorMessage = yield ApiError.errorToHumanString(error);
+    yield put(setErrorMessage(errorMessage));
+  } finally {
+    yield put(hideLoading());
+  }
+}
+
+export const TAXONOMY_TERMS_REQUESTED = 'TAXONOMY_TERMS_REQUESTED';
+export const requestTaxonomyTerms = taxonomyVocabulary => ({
+  type: TAXONOMY_TERMS_REQUESTED,
+  payload: { taxonomyVocabulary },
+});
+
+export const getTaxonomyVocabularyCache = state =>
+  state.taxonomy.taxonomyVocabulary;
+export const getTaxonomyVocabularyById = (taxonomyVocabularyList, vocabulary) =>
+  taxonomyVocabularyList
+    .filter(({ attributes: { vid } }) => vid === vocabulary)
+    .shift();
+
+export const TAXONOMY_TERMS_LOADED = 'TAXONOMY_TERMS_LOADED';
+function* loadTaxonomyTerms(action) {
+  try {
+    const {
+      payload: { taxonomyVocabulary },
+    } = action;
+    yield put(resetLoading());
+    yield put(showLoading());
+
+    const { data: taxonomyTerms } = yield call(api, 'taxonomy_term', {
+      parameters: { type: taxonomyVocabulary },
+    });
+
+    const vocabulary = yield select(getTaxonomyVocabularyCache);
```

Comment:
May I ask why we not just make ```getTaxonomyVocabularyById``` get the full state passed in? This way we wouldn't need two steps.